(DOCSP-14643): Port Swift FAQ info from Realm.io (#978)
mongodb/docs-realm: source/sdk/ios/fundamentals/object-models-and-schemas.txt

   To learn how to add relationships to your Realm object, see
   :ref:`ios-declare-relationship-properties`.

> Realm Does Not Support Swift Structs

Author: It does. And I don't like a negative header in general. I've shortened it to just "Structs." What do you think?

> Realm Does Not Support Swift Structs

Reviewer: I can't currently check the staged build due to https://jira.mongodb.org/browse/MARKETING-21151, but I think the name is fine.

> Realm Does Not Support Swift Structs

Reviewer: You might want to specify "Swift Structs". Obj-C objects can't go in (plain C) structs in Obj-C, so it's kind of implied.

------------------------------------

{+client-database+} does not currently support structs as models for a 

> variety of reasons. Foremost, {+service-short+}'s design focuses on 

Reviewer: Word choice:
```suggestion
variety of reasons. {+service-short+}'s design focuses on 
```

“live” objects. This concept is not compatible with value type structs. 

> By design, {+service-short+} provides features which are incompatible with 

Reviewer: ```suggestion
By design, {+service-short+} provides features that are incompatible with 
```

these semantics, such as:
 

> - Live data

Reviewer: Link to live objects page?

> - Reactive APIs

Reviewer: Link to react to changes page?

- Low memory footprint of data
- Good operation performance

> - Lazy and cheap access to partial data

Reviewer: Live queries link?

> - Lack of data serialization/deserialization

Reviewer: Link to Object Models & Schemas?

Author: This content is on the Object Models & Schemas page. Is there another part of the page where you think this should link, @cbush ?

Reviewer: Oh right. I was thinking about the intro paragraph. But there's not a specific place where we talk about the fact that you don't need an ORM, etc. for Realm. Oh well.

> - Keeping potentially complex object graphs synchronized

Reviewer: Link to Sync?

That said, it is sometimes useful to detach objects from their backing 
{+realm+}. This typically isn't an ideal design decision. Instead, 
developers use this as a workaround for temporary limitations in our 
library. 

> To make this easier, we’ve implemented standalone/detached {+service-short+} 

Reviewer: ```suggestion
To make this easier, we've implemented standalone/detached {+service-short+} 
```

> objects to behave like :apple:`Apple's NSObject <documentation/objectivec/nsobject>`. 

Reviewer: "Implemented standalone/detached ... to behave" reads a bit weird. How about:

>We've made unmanaged objects behave like ..."

> We support making “in-memory copies” of {+service-short+} objects. Use an initializer 

Reviewer: Watch out for smart quotes:
```suggestion
We support making "in-memory copies" of {+service-short+} objects. Use an initializer 
```

------------------------------------

{+client-database+} does not currently support structs as models for a 
variety of reasons. {+service-short+}'s design focuses on “live” objects. 
This concept is not compatible with value type structs. By design, 
{+service-short+} provides features that are incompatible with these 
semantics, such as:
 
- :ref:`Live data <ios-live-object>`
- :ref:`Reactive APIs <ios-react-to-changes>`
- Low memory footprint of data
- Good operation performance
- :ref:`Lazy and cheap access to partial data <ios-live-queries>`
- Lack of data serialization/deserialization
- :ref:`Keeping potentially complex object graphs synchronized <ios-sync-changes-between-devices>`

That said, it is sometimes useful to detach objects from their backing 
{+realm+}. This typically isn't an ideal design decision. Instead, 
developers use this as a workaround for temporary limitations in our 
library. 

To make this easier, we've made unmanaged objects behave like 
:apple:`Apple's NSObject <documentation/objectivec/nsobject>`. Use an 

> initializer to create an unmanaged object by key-value coding copying a 

Reviewer: Can we simplify this?

>Create an unmanaged copy by passing the managed instance to the object's initializer.

Author: I've [rewritten this paragraph](https://github.com/mongodb/docs-realm/pull/978/commits/e5008ff2a529e0be704fc0c1ed6a9b5d8d60dd33#diff-ea0e01aa4cd7351e0e98d04b778142d1e208bf07da90437e9cd585ce922afed7R136) a bit. What do you think, @cbush ?

Reviewer: Perfect.

- You can define to-one, to-many, and inverse relationships in your schema.
  Inverse relationships automatically form backlinks.

- {+service-short+} does not currently support structs. When required, 

>   create "in-memory copies" of {+service-short+} objects.

Reviewer: Does this make it sound more complicated than it is? I feel like the point here is that you can use "unmanaged" objects like normal objects. (Standalone, in-memory, detached are all new words for "unmanaged")
